bin/express-cli.js

```diff
@@ -325,15 +376,27 @@ function createApplication (name, dir) {
   if (program.git) {
     copyTemplate('js/gitignore', path.join(dir, '.gitignore'))
   }
+  if (program.typescript) {
+    copyTemplate('ts/tsconfig.json', path.join(dir, 'tsconfig.json'))
+  }
 
   // sort dependencies like npm(1)
   pkg.dependencies = sortedObject(pkg.dependencies)
 
   // write files
-  write(path.join(dir, 'app.js'), app.render())
+  var appScriptName
+  var wwwScriptName
+  if (program.typescript) {
+    appScriptName = 'app.ts'
+    wwwScriptName = 'bin/www.ts'
```

Comment:
The `www` script shouldn't have an extension; it follows the pattern of being an executable (with shebang on the top).

Reply:
Got it, may be adding a post-build script to create a symbolic link with name www, pointing to www .js?

As some people may want to generate source-maps for the compiled JS files, I afraid directly renaming it may break the file linkage.

For example,

```js
// add postbuild.js

var fs = require('fs')
var path = require('path')

// Create a symlink, linking www to www.js
var wwwScriptPath = path.join(__dirname, 'bin/www.js')
var wwwPath = path.join(__dirname, 'bin/www')
if (fs.existsSync(wwwScriptPath) && !fs.existsSync(wwwPath)) {
  fs.symlinkSync(wwwScriptPath, wwwPath)
}
```

Reply:
Unfortunately normal users cannot create symbolic links on windows.

Reply:
How about if coping www .js as www?

```js
var fs = require('fs')
var path = require('path')

// Polyfill copyFileSync for node < 8.5.0
if (!fs.copyFileSync) {
  fs.copyFileSync = function (src, dest, flags) {
    fs.writeFileSync(dest, fs.readFileSync(src), flags)
  }
}

// Copy www.js to www
var wwwScriptPath = path.join(__dirname, 'bin/www.js')
var wwwPath = path.join(__dirname, 'bin/www')
if (fs.existsSync(wwwScriptPath)) {
  fs.copyFileSync(wwwScriptPath, wwwPath)
}
```